### PR TITLE
Add min/max values to the statistics plots

### LINKED ISSUE #133: Min/max plot
Create a radio button to show the min/max and mean values in addition to statistics and raw data. The min and max are the highest and lowest realization (final data point in the objective function evaluation).

### PR DESCRIPTION
**Pull request by verveerpj**
Adds the min/max values to the statistics plot, while also showing the corresponding realization when hovering over the min/max points.

closes #133

### CODE REVIEW

File: everviz/plugins/summary_plot/summary_callback.py [hunk 1]
```diff
@@ -60,6 +60,30 @@ def _get_statistics_lines(
...
                     ),
+                    go.Scatter(
+                        y=line_data["min_value"],
+                        x=line_data[x_key],
+                        mode="markers",
+                        marker={"size": 10, "symbol": "square"},
+                        line={"color": color},
+                        name=key,
+                        showlegend=False,
+                        hovertext=[
```
Comment (DanSava) [inline everviz/plugins/summary_plot/summary_callback.py:71]: Maybe modify the hover text similarly to the suggestion for the objectives plot

File: everviz/plugins/summary_plot/summary_callback.py [hunk 2]
```diff
@@ -60,6 +60,30 @@ def _get_statistics_lines(
...
+                    ),
+                    go.Scatter(
+                        y=line_data["max_value"],
+                        x=line_data[x_key],
+                        mode="markers",
+                        marker={"size": 10, "symbol": "square"},
+                        line={"color": color},
+                        name=key,
+                        showlegend=False,
+                        hovertext=[
```
Comment (DanSava) [inline everviz/plugins/summary_plot/summary_callback.py:83]: Same as above

File: everviz/plugins/summary_plot/summary_callback.py [hunk 3]
```diff
@@ -73,32 +97,35 @@ def _get_data_lines(data, summary_key_list, line_list, x_filter, x_key, mode="ma
...
+            # Plot each realization separately, adapting the hovertext.
+            for real in realizations:
+                line = line_data[line_data["realization"].isin([real])]
+                traces.append(
+                    go.Scatter(
+                        y=line[key],
+                        x=line[x_key],
+                        mode=mode,
+                        marker={"size": 10},
+                        name=name,
```
Comment (DanSava) [inline everviz/plugins/summary_plot/summary_callback.py:123]: If the name is to appear in the legend it should contain the realization number 
```
name  = f"{name} realization {real}"
```

File: everviz/plugins/summary_plot/summary_callback.py [hunk 4]
```diff
@@ -73,32 +97,35 @@ def _get_data_lines(data, summary_key_list, line_list, x_filter, x_key, mode="ma
...
+                        showlegend=True,
+                        line={"color": color},
+                        hovertext=f"realization:{real}",
```
Comment (DanSava) [inline everviz/plugins/summary_plot/summary_callback.py:126]: Is this really needed if the legend contains the realization number? If yes then it should also contain the batch number and the key.

**Review by DanSava — APPROVED** [review]
Looks good had just some renaming suggestions

**Comment by verveerpj** [comment]
Considering you remarks about the hover and legend text, I have revised as follows: realization is added to the legend, if that is possible. If no legend is shown, it is added to the hover text. They key is not added to the hover text, because it shows the name of the line already, which is generally the key..

**Pull request merged**